integration/oci_calls/oci_call_test.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will verify the arguments of
+# running or stopping a container matches
+# with the OCI calls
+
+set -e
+
+cidir=$(dirname "$0")
+source "${cidir}/../../lib/common.bash"
+
+# Save logs
+TMP_FILE=$(mktemp runtimelogs.XXXXX)
+# Environment variables
+IMAGE="busybox"
+PAYLOAD="tail -f /dev/null"
+NAME="testoci"
+
+function remove_tmp_file {
+	rm -rf $TMP_FILE
+}
+
+trap remove_tmp_file EXIT
+
+# Get time to filter the logs
+function get_time() {
+	start_time=$(date "+%F %H:%M:%S")
+}
+
+# Get log for a specific time
+function get_debug_logs() {
+	sudo journalctl -q --since "$start_time" -o cat -a -t ${RUNTIME} > ${TMP_FILE}
+}
+
+# Find the arguments or oci calls for a specific command
+function check_arguments() {
+	list_arguments=$(grep -o "arguments=[^ ]*" ${TMP_FILE} --color|cut -d= -f2-|tr -d '"'|tr -d "\\\\")
+
+	[ -n "$list_arguments" ] || die "List of arguments missing"
+
+	# Check arguments vs oci calls
+	for i in "${oci_call[@]}"; do
+		echo "$list_arguments" | grep -w "$i" > /dev/null
+	done
+}
+
+# Find the order of the arguments is equal to the order of the oci call
+function order_arguments() {
+	local -a final_arguments=$(echo ${list_arguments//state/}|awk '{for (i=1;i<=NF;i++) if (!a[$i]++) printf("%s%s",$i,FS)}'|sed 's/ *$//')
```

Comment:
Hi @GabyCT - please can you format this awk script across multiple lines, add some more braces for clarity along with comments explaining how this is working as it's rather terse atm! ;)